### DISCUSSION TITLE
Questions about multiple windows

### DISCUSSION THREAD

**Discussion by NeatSketch** [google/filament#6866] [Q&A] [answered]
Hello!

Regarding multi-windowing:

1. Can textures, materials, & meshes be shared between Renderers?
More generally, can I use the same textures, materials, & meshes in two windows without loading them twice?

2. Can windows be rendered on separate threads?

3. Can both of the above be done at the same time?

**Comment by romainguy** [comment, marked as answer]
1. Yes. All resources in the same Engine are shared across Renderer instances. This includes scenes, meshes, textures, materials, views etc.

2. No, once an Engine is created all APIs for that Engine must be invoked from the same thread. Filament itself is heavily multi-threaded though. You can however create multiple Engines (but you won't share resources).

**Reply by NeatSketch** [reply]
Thank you.
Maybe there is another way to synchronize rendering of two windows with two different displays with different refresh rates besides using two threads for each window?
